Tests/Integration.Tests.ps1

```diff
@@ -0,0 +1,97 @@
+Import-Module -Force $PSScriptRoot/../Docker.Build.psm1
+. "$PSScriptRoot\..\Private\Invoke-Command.ps1"
+. "$PSScriptRoot\..\Private\Utilities.ps1"
+
+Describe 'Use cases for this module' {
+
+    Context 'With a local docker registry with auth required' {
+
+        $localRegistryName = 'localhost:5000'
+        $dockerImageNamePrefix = 'integration-testcase'
+
+        BeforeAll {
+            $testData = Join-Path (Split-Path -Parent $PSScriptRoot) "Test-Data"
+            $htpasswdPath = Join-Path $testData 'DockerRegistry'
+            $dockerImages = Join-Path $testData 'DockerImage'
+            $exampleRepos = Join-Path $testData 'ExampleRepos'
+            $removeImageCommand = 'docker image rm --force localhost:5000/integration-testcase-2:latest'
+            $pruneImageCommand = 'docker system prune --force'
+            $startRegistryCommand = "docker run -d -p 5000:5000 --name registry -v ${htpasswdPath}:/auth -e 'REGISTRY_AUTH=htpasswd' -e 'REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm' -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd registry:2"
+            Invoke-Command $removeImageCommand
+            Invoke-Command $pruneImageCommand
+            Invoke-Command $startRegistryCommand
+        }
+
+        AfterAll {
+            Invoke-Command 'docker container stop registry'
+            Invoke-Command 'docker container rm -v registry'
+        }
+
+        BeforeEach {
+            $script:backupLocation = Get-Location
```

Comment:
With `Push-Location` and `Pop-Location`, this line is not needed.

Reply:
Push/Pop location is a stack of paths that one can use to go deep into different paths without having to remember its previous path. This mechanism ensures that wherever you go to, you will always be able to get back to where you start. 

When you have no more path to pop, `Pop-Location` does nothing.